src/Compilers/Core/Portable/StrongName/DesktopStrongNameProvider.cs

```diff
@@ -131,63 +136,22 @@ public DesktopStrongNameProvider(ImmutableArray<string> keyFileSearchPaths) : th
         /// </summary>
         /// <param name="tempPath">Path to use for any temporary file generation.</param>
         /// <param name="keyFileSearchPaths">An ordered set of fully qualified paths which are searched when locating a cryptographic key file.</param>
-        public DesktopStrongNameProvider(ImmutableArray<string> keyFileSearchPaths = default(ImmutableArray<string>), string tempPath = null)
+        public DesktopStrongNameProvider(ImmutableArray<string> keyFileSearchPaths = default(ImmutableArray<string>), string tempPath = null): this(keyFileSearchPaths, tempPath, null)
+        {
+        }
+
+        internal DesktopStrongNameProvider(ImmutableArray<string> keyFileSearchPaths, string tempPath, StrongNameFileSystem strongnamefileSystem)
         {
             if (!keyFileSearchPaths.IsDefault && keyFileSearchPaths.Any(path => !PathUtilities.IsAbsolute(path)))
             {
                 throw new ArgumentException(CodeAnalysisResources.AbsolutePathExpected, nameof(keyFileSearchPaths));
             }
 
+            FileSystem = strongnamefileSystem ?? new StrongNameFileSystem();
             _keyFileSearchPaths = keyFileSearchPaths.NullToEmpty();
             _tempPath = tempPath;
         }
 
-        internal virtual bool FileExists(string fullPath)
-        {
-            Debug.Assert(fullPath == null || PathUtilities.IsAbsolute(fullPath));
-            return File.Exists(fullPath);
-        }
-
-        internal virtual byte[] ReadAllBytes(string fullPath)
-        {
-            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
-            return File.ReadAllBytes(fullPath);
-        }
-
-        /// <summary>
-        /// Resolves assembly strong name key file path.
-        /// Internal for testing.
-        /// </summary>
-        /// <returns>Normalized key file path or null if not found.</returns>
-        internal string ResolveStrongNameKeyFile(string path)
```

Comment:
Did this method need to move to `StrongNameFileSystem`? `StrongNameFileSystem` would be simpler if this method stays here and callers of this method would not need to change.

Reply:
This method would need to be copied to DesktopStrongNameProvider _and_ PortableStrongNameProvider then.

Reply:
I see, thanks. In that case, it makes more sense to move the method.